File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsaccountpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsAccountPagedResponseHrefModel } from "./codatdatacontractsdatasetsaccountpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsAccountPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsAccountPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsAccountPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsAccountPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsAccountPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsAccountPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsAccountPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsAccountPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsAccountPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_accountpagedresponselinksmodel.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_accountpagedresponsehrefmodel as shared_codat_datacontracts_datasets_accountpagedresponsehrefmodel
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsAccountPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_accountpagedresponsehrefmodel.CodatDataContractsDatasetsAccountPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current'), 'exclude': lambda f: f is None }})
    next: Optional[shared_codat_datacontracts_datasets_accountpagedresponsehrefmodel.CodatDataContractsDatasetsAccountPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next'), 'exclude': lambda f: f is None }})
    previous: Optional[shared_codat_datacontracts_datasets_accountpagedresponsehrefmodel.CodatDataContractsDatasetsAccountPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous'), 'exclude': lambda f: f is None }})
    self: Optional[shared_codat_datacontracts_datasets_accountpagedresponsehrefmodel.CodatDataContractsDatasetsAccountPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self'), 'exclude': lambda f: f is None }})
    